code/modules/tgui/tutorial/tgui_template_machine.dm

```diff
@@ -0,0 +1,62 @@
+/obj/machinery/tgui_template
+	name = "tgui template machine"
+	desc = "Coders' helper."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "sextractor_open"
+
+	var/list/options
+	var/ui_message
+	var/current_option
+
+/obj/machinery/tgui_template/Initialize(mapload)
+	. = ..()
+
+	options = list("Love", "Bravery", "Honor")
+	ui_message = "Hello, world!"
+	current_option = options[1]
+
+/obj/machinery/tgui_template/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "TGUITemplate")
+		ui.open()
+
+
+/obj/machinery/tgui_template/ui_data(mob/user)
+	var/list/data = list()
+	. = data
+
+	data["options"] = options
+	data["ui_message"] = ui_message
+	data["current_option"] = current_option
+
+/obj/machinery/tgui_template/ui_static_data(mob/user)
+	var/list/data = list()
+	. = data
+
+	data["something_static"] = "This is static string. This is never changed."
+	// if you really need update, use update_static_data() or update_static_data_for_all_viewers() procs
+
+/obj/machinery/tgui_template/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/research_designs)) // sample
+
+/obj/machinery/tgui_template/ui_act(action, params)
+	if(..())
```

Comment:
this is nonstandard

`. = ..(), if(.) return` is standard

Reply:
Assigning to . first is important since if we return we need to indicate that we want the UI to update.